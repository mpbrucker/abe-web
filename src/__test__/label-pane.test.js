import React from 'react';
import renderer from 'react-test-renderer';
import LabelPane from '../components/label-pane';

describe('LabelPane', () => {
    const labels = {
        first: { color: 'red', description: 'first label' },
        second: { color: 'green', description: 'second label' }
    };
    test('matches snapshot', () => {
        const component = renderer.create(
            <LabelPane
                contentClass="css-class"
                general={{}}
                possibleLabels={labels}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('with selected labels', () => {
        const component = renderer.create(
            <LabelPane
                contentClass="css-class"
                general={{}}
                editable
                possibleLabels={labels}
                selectedLabels={['first']}
            />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('editable', () => {
        const component = renderer.create(
            <LabelPane
                contentClass="css-class"
                general={{}}
                editable
                possibleLabels={labels}
            />
        );
        //  selectedLabels  labelToggled editable
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
