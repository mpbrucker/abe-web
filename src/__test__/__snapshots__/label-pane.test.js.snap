// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelPane matches snapshot 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}.label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button label first"
      id="label-first"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      first
    </button>
    <button
      className="button label second"
      id="label-second"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      second
    </button>
  </div>
</div>
`;

exports[`LabelPane not editable 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}.label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <span
      className="label first selected"
      id="label-first"
      title="first label"
    >
      <ion-icon
        name="pricetag"
      />
      first
    </span>
    <span
      className="label second"
      id="label-second"
      title="second label"
    >
      <ion-icon
        name="pricetag"
      />
      second
    </span>
  </div>
</div>
`;

exports[`LabelPane with selected labels 1`] = `
<div
  className="css-class"
>
  <style
    type="text/css"
  >
    .label.button.first:not(.no-hover):hover,.label.first.selected{background-color:red;}.label.button.second:not(.no-hover):hover,.label.second.selected{background-color:green;}
  </style>
  <div
    className="label-selector-list"
  >
    <button
      className="button label first selected"
      id="label-first"
      onClick={[Function]}
      title="first label"
      type="button"
    >
      first
    </button>
    <button
      className="button label second"
      id="label-second"
      onClick={[Function]}
      title="second label"
      type="button"
    >
      second
    </button>
  </div>
</div>
`;
